orangecontrib/text/corpus.py

```diff
@@ -100,6 +100,30 @@ def extend_corpus(self, metadata, Y):
         self.X = self.W = np.zeros((len(self), 0))
         Table._init_ids(self)
 
+    def update_attributes(self, X, dictionary):
+        """
+        Adds X to the corpus and updates its domain accordingly.
+        """
+        self.X = X
```

Comment:
Shouldn't this be stacked with the previous X? What if X had some attributes before, they are not lost.